cmd/maesh/maesh.go

```diff
@@ -106,32 +111,72 @@ func maeshCommand(iConfig *cmd.MaeshConfiguration) error {
 	}
 
 	aclEnabled := iConfig.ACL || iConfig.SMI
-
 	log.Debugf("ACL mode enabled: %t", aclEnabled)
 
-	// Create a new stop Channel
-	stopCh := signals.SetupSignalHandler()
-	// Create a new ctr.
+	apiServer, err := api.NewAPI(log, iConfig.APIPort, iConfig.APIHost, clients.KubernetesClient(), iConfig.Namespace)
+	if err != nil {
+		return fmt.Errorf("unable to create the API: %w", err)
+	}
+
 	ctr, err := controller.NewMeshController(clients, controller.Config{
 		ACLEnabled:       aclEnabled,
 		DefaultMode:      iConfig.DefaultMode,
 		Namespace:        iConfig.Namespace,
 		IgnoreNamespaces: iConfig.IgnoreNamespaces,
-		APIPort:          iConfig.APIPort,
-		APIHost:          iConfig.APIHost,
 		MinHTTPPort:      minHTTPPort,
 		MaxHTTPPort:      minHTTPPort + iConfig.LimitHTTPPort,
 		MinTCPPort:       minTCPPort,
 		MaxTCPPort:       minTCPPort + iConfig.LimitTCPPort,
 		MinUDPPort:       minUDPPort,
 		MaxUDPPort:       minUDPPort + iConfig.LimitUDPPort,
-	}, log)
+	}, apiServer, log)
 	if err != nil {
 		return fmt.Errorf("unable to create controller: %w", err)
 	}
 
-	// run the ctr loop to process items
-	ctr.Run(stopCh)
+	errCh := make(chan error, 2)
+
+	var wg sync.WaitGroup
```

Comment:
Thought, is it really needed to run the `api` and the `controller` in a go routine?

Reply:
I think it is. At least one of them must be run in a goroutine as `api.ListenAndServe` and `ctrl.Start` are blocking. Then if we don't run both in a goroutine, if the API stops with an error we won't to be able to stop the controller.
So the two options I see are:
- Run the `ListenAndServe` and `Start` in 2 goroutine.
- Or just run the stop mechanism in a goroutine.